go/cbrotli.go

```diff
@@ -0,0 +1,264 @@
+// Copyright 2016 Google Inc. All Rights Reserved.
+//
+// Distributed under MIT license.
+// See file LICENSE for detail or copy at https://opensource.org/licenses/MIT
+
+// Package cbrotli compresses and decompresses data with C-Brotli library.
+package cbrotli
+
+import (
+	"bytes"
+	"errors"
+	"io"
+
+	"github.com/google/brotli/go/cgo/decoder"
+	"github.com/google/brotli/go/cgo/encoder"
+)
+
+//------------------------------------------------------------------------------
+// Encoder
+//------------------------------------------------------------------------------
+
+// WriterOptions configures Writer.
+type WriterOptions struct {
+	// Quality controls the compression-speed vs compression-density trade-offs.
+	// The higher the quality, the slower the compression. Range is 0 to 11.
+	Quality int
+	// LGWin is the base 2 logarithm of the sliding window size.
+	// Range is 10 to 24. 0 indicates automatic configuration based on Quality.
+	LGWin int
+}
+
+// ErrEncodeFailure is returned when the encode operation fails.
+var ErrEncodeFailure = errors.New("Brotli encode failure")
```

Comment:
We should not export these errors. Singleton error values are generally not a good design pattern. 

My recommendation is to not export these and add them later if someone needs them. Once added, the API is hard to change once users are depending on it.